HACKING.md

```diff
@@ -31,58 +31,103 @@ echo 'logger custom post reboot script ran' > /etc/auter/post-reboot.d/post_rebo
+
+[ pass/fail ] auter --apply
+    - Check: prints "INFO: Running with: /usr/bin/auter --apply; Applying updates; Updates complete, you may need to reboot for some updates to take effect
+    - Check: expected updates were applied using 'yum history info' or 'dnf history info'
+    - Check: /var/lib/auter/last-update-default contains update info
+    - Check: no upates available after running
+    - Check: pre/post apply scripts ran successfully, messages logged to syslog
+
+[ pass/fail ] auter --reboot working
+    - Check: reboot scheduled in 2 minutes time
+    - Check: prints "INFO: Running with: ./auter --reboot; Rebooting server" followed by shutdown message
+    - Check: 5 minutes after reboot is complete pre/post reboot scripts ran successfully with messages logged to syslog
 
-**Release Process**
+#### updates via cron
 
-1.  Ensure that all files are updated to the version of the release number you're about to tag.  This includes:
-  - auter
-  - NEWS
-  - auter.spec
+[ pass/fail ] auter --prep --stdout
+    - Check: updates downloaded to /var/cache/auter/default
+    - Check: pre/post prep scripts ran successfully, messages logged to syslog
+    - Check: output from auter also mailed to root user on CentOS boxes, output logged to syslog on Fedora systems
 
-2.  Add notes to auter.spec and NEWS with a list of the changes for this release
+[ pass/fail ] auter --apply --stdout
+    - Check: no updates available after running
+    - Check: pre/post scripts ran successfully with messages logged to syslog
+    - Check: output from auter also mailed to root user on CentOS boxes, output logged to syslog on Fedora systems
+
+[ pass/fail ] auter --reboot --stdout
+    - Check: server rebooted after 2 minutes
+    - Check: pre/post scripts ran successfully
+    - Check: output from auter also mailed to the root user on CentOS, output logged to syslog on Fedora
+
+#### new functionality testing
+
+[ pass/fail ] <test new functionality added>
+    - Check:
+```
+
+#### Documentation
+
+When making any changes to code, make sure documentation (--help, man page) has been updated with the new functionality.
+
+#### Merge Rules
+
+- All pull requests should be made against the develop branch
+- All pull requests MUST be reviewed and approved before merging
+- Pull request reviews and merges MUST be completed by another maintainer
+- If possible squash and merge commits
+- All travis-ci tests should pass before merging
+    - If the spellcheck test fails, adding the problematic words to the dictionary is a valid option
+    - If the shellcheck test fails, you should fix the issues mentioned or adjust the test with explicit comments for the check code and why it is being ignored
+    - The adjustments to the test files may need to be merged before the PR will pass the tests. This should be done as a separate PR referencing the associated PR for the changes.
 
-3.  Push to github
+#### Release Process
```

Comment:
This process has changed. When a release needs to be done the following process should be followed:

1) A new issue should be raised with the title of "Prep for <VERSION> Release"
2) Ensure all required PRs for the new release have been merged to the develop branch
3) A maintainer with admin rights to the master branch in rackerlabs/auter should create a new branch based from the develop branch.:
```
git clone --branch=develop https://github.com/rackerlabs/auter.git rackerlabs/auter
cd rackerlabs/auter
git checkout -b Release-<NEW_VERSION>
git push origin Release-<NEW_VERSION>
```
4) Any new PRs should be made to the develop branch. The only changes to the Release-<NEW_VERSION> should be fixing of any review issues for that branch.

5) Full testing for all supported OSs should be carried out. Either as delegated tasts traced in a github project. Example: https://github.com/rackerlabs/auter/projects/1

6) Once all testing has been completed for Release-<NEW_VERSION> The reviewer should merge the Release-<NEW_VERSION> branch to both master and develop branches.

7) Tag a new release named <NEW_VERSION> (Eg: Release 0.11)

8) ... These steps will need to be documented at a later date but this will be the process to push the new release to EPEL and Fedora repos